### PR TITLE
windows path seperator fix

### LINKED ISSUE #3389: Test script fails due to path separator // in registry/directory on windows platform
/kind bug

## What versions of software are you using?

**Operating System:**
Win10

**Output of `odo version`:**
master

## How did you run odo exactly?

Ran `make test-e2e-images` on windows.

## Actual behavior

```
> make test-e2e-images
[...]
SSSSSSSCreated dir: C:\Users\Admin\AppData\Local\Temp\108651876
Creating a new project: asigcqzcof
Running odo.exe with args [odo project create asigcqzcof -w -v4]
[odo]  -  Waiting for project to come up  ...
[odo] I0619 06:50:12.739836    3816 occlient.go:543] Status of creation of project asigcqzcof is Active
[odo] I0619 06:50:12.788612    3816 occlient.go:548] Project asigcqzcof now exists
[odo] I0619 06:50:12.827575    3816 occlient.go:583] Status of creation of service account &ServiceAccount{ObjectMeta:{default  asigcqzcof /api/v1/namespaces/asigcqzcof/serviceaccounts/default 10fa5fca-0585-4a1b-b823-a84cdc72fbf3 103419 0 2020-06-19 06:50:13 +0000 GMT <nil> <nil> map[] map[] [] []  []},Secrets:[]ObjectReference{ObjectReference{Kind:,Namespace:,Name:default-token-6289w,UID:,APIVersion:,ResourceVersion:,FieldPath:,},ObjectReference{Kind:,Namespace:,Name:default-dockercfg-qmhhx,UID:,APIVersion:,ResourceVersion:,FieldPath:,},},ImagePullSecrets:[]LocalObjectReference{LocalObjectReference{Name:default-dockercfg-qmhhx,},},AutomountServiceAccountToken:nil,} is ready
 V  Waiting for project to come up [808ms]
[odo]  V  Project 'asigcqzcof' is ready for use
[odo]  V  New project created and now using project: asigcqzcof
[odo] I0619 06:50:12.852064    3816 odo.go:72] Could not get the latest release information in time. Never mind, exiting gracefully :)



Running oc.exe with args [oc --request-timeout 5m import-image nodejs:latest --namespace=asigcqzcof 
--from=registry.access.redhat.com\rhscl\nodejs-8-rhel7:latest --confirm]

[oc] The ImageStreamImport "nodejs" is invalid: spec.images[0].from.name: Invalid value: 
"registry.access.redhat.com\\rhscl\\nodejs-8-rhel7:latest": invalid reference format



Deleting project: asigcqzcof
Running odo.exe with args [odo project delete asigcqzcof -f]
[odo] I0619 06:50:13.864599   10320 application.go:49] Unable to list Service Catalog instances: unable to list ServiceInstances: the server could not find 
the requested resource (get serviceinstances.servicecatalog.k8s.io)
[odo]  V  Deleted project : asigcqzcof
[odo]  !  Warning! Projects are deleted from the cluster asynchronously. Odo does its best to delete the project. Due to multi-tenant clusters, the project 
may still exist on a different node.
[odo] I0619 06:50:13.989486   10320 odo.go:72] Could not get the latest release information in time. Never mind, exiting gracefully :)
Deleting dir: C:\Users\Admin\AppData\Local\Temp\108651876
------------------------------
+ Failure [2.245 seconds]
odo supported images e2e tests
C:/Users/Admin/go/src/github.com/openshift/odo/tests/e2escenarios/e2e_images_test.go:14
  odo supported images deployment
  C:/Users/Admin/go/src/github.com/openshift/odo/tests/e2escenarios/e2e_images_test.go:83
    Should be able to verify the nodejs-8-rhel7 image [It]
    C:/Users/Admin/go/src/github.com/openshift/odo/tests/e2escenarios/e2e_images_test.go:94
    No future change is possible.  Bailing out early after 0.378s.
    Running oc.exe with args [oc --request-timeout 5m import-image nodejs:latest --namespace=asigcqzcof --from=registry.access.redhat.com\rhscl\nodejs-8-rhel7:latest --confirm]
    Expected
        <int>: 1
    to match exit code:
        <int>: 0
    C:/Users/Admin/go/src/github.com/openshift/odo/tests/helper/helper_run.go:34
------------------------------
[...]
SSS
Summarizing 8 Failures:
[Fail] odo supported images e2e tests odo supported images deployment [It] Should be able to verify the nodejs-8-rhel7 image 
C:/Users/Admin/go/src/github.com/openshift/odo/tests/helper/helper_run.go:34
[...]
Ran 8 of 23 Specs in 20.037 seconds
FAIL! -- 0 Passed | 8 Failed | 0 Pending | 15 Skipped
--- FAIL: TestE2eScenarios (20.52s)
FAIL
```

Failure : The ImageStreamImport "nodejs" is invalid: spec.images[0].from.name: Invalid value: "registry.access.redhat.com\\rhscl\\nodejs-8-rhel7:latest": invalid reference format

## Expected behavior

Image import path should be `registry.access.redhat.com\rhscl\nodejs-8-rhel7:latest`

### LINKED ISSUE #4235: E2e images test script fails while image verification on windows platform
/kind bug

## What versions of software are you using?

**Operating System:**  Win10

**Output of `odo version`:**
master

## How did you run odo exactly?
```>make test-e2e-images``` 

## Actual behavior
Test is failing on verifying images in e2e test on windows10

## Expected behavior
Test should pass on win10

## Any logs, error output, etc?

```[2] [odo] [WARNING] Configuration options: 'appendAssemblyId' is set to false, and 'classifier' is missing.
[2] [odo] Instead of attaching the assembly file: /tmp/src/target/javalin-hello-world-0.1-SNAPSHOT.jar, it will become the file for main project artifact.
[2] [odo] NOTE: If multiple descriptors or descriptor-formats are provided for this project, the value of this file will be non-deterministic!
[2] [odo] [WARNING] Replacing pre-existing project main-artifact file: /tmp/src/target/javalin-hello-world-0.1-SNAPSHOT.jar
[2] [odo] with assembly file: /tmp/src/target/javalin-hello-world-0.1-SNAPSHOT.jar
[2] [odo] [INFO] ------------------------------------------------------------------------
[2] [odo] [INFO] BUILD SUCCESS
[2] [odo] [INFO] ------------------------------------------------------------------------
[2] [odo] [INFO] Total time:  4.214 s
[2] [odo] [INFO] Finished at: 2020-11-18T13:16:34Z
[2] [odo] [INFO] ------------------------------------------------------------------------
[2] [odo] [WARNING] The requested profile "openshift" could not be activated because it does not exist.
[2] [odo] INFO Copying deployments from target to /deployments...
[2] [odo] '/tmp/src/target/javalin-hello-world-0.1-SNAPSHOT.jar' -> '/deployments/javalin-hello-world-0.1-SNAPSHOT.jar'
[2] [odo] + '[' '!' -z /home/jboss ']'
[2] [odo] + '[' -n /opt/app-root/src-backup ']'
[2] [odo] + '[' /tmp '!=' /home/jboss ']'
[2] [odo] + '[' /deployments '!=' /home/jboss ']'
[2] [odo] + rsync -rlO /opt/app-root/src-backup/src/. /tmp/src/
[2] [odo] + '[' -z '' ']'
[2] [odo] + PV_MNT_PT=/opt/app-root
[2] [odo] + sed -i s/autostart=false/autostart=true/g /opt/app-root/conf/supervisor.conf
[2] [odo] + /opt/odo/bin/supervisord ctl stop run
[2] [odo] + /opt/odo/bin/supervisord ctl start run
[2] [odo] run: stopped
[2] [odo] run: started
 V  Building component [8s]
[2] [odo]  V  Changes successfully pushed to component
[2] [odo] I1118 13:16:36.220633    5080 odo.go:72] Could not get the latest release information in time. Never mind, exiting gracefully :)
[2] Checking http://openjdk-app-8080-app-hlongqcuop.apps.ci-ln-jyvbx3b-d5d6b.origin-ci-int-aws.dev.rhcloud.com, for Hello Java UPDATED
[2] try 0 of 90
[2] Running command:  [odo watch -v 4 --context C:\Users\Admin\AppData\Local\Temp\192507358]
[2] Running curl.exe with args [curl http://openjdk-app-8080-app-hlongqcuop.apps.ci-ln-jyvbx3b-d5d6b.origin-ci-int-aws.dev.rhcloud.com]
[2] Deleting dir: C:\Users\Admin\AppData\Local\Temp\192507358\abcd
[2] Replacing "Hello" with "Hello odo" in C:\Users\Admin\AppData\Local\Temp\192507358\src\main\java\MessageProducer.java
[2] [curl]   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
[2] [curl]                                  Dload  Upload   Total   Spent    Left  Speed
100    38  100    38    0     0     38      0  0:00:01 --:--:--  0:00:01   304
[2] [curl] Hello Java UPDATED World from Javalin!Running curl.exe with args [curl http://openjdk-app-8080-app-hlongqcuop.apps.ci-ln-jyvbx3b-d5d6b.origin-ci-int-aws.dev.rhcloud.com]
[2] [curl]   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
[2] [curl]                                  Dload  Upload   Total   Spent    Left  Speed
100    42  100    42    0     0     42      0  0:00:01 --:--:--  0:00:01   206
[2] [curl] Hello odo Java UPDATED World from Javalin!Running oc.exe with args [oc get pods --namespace hlongqcuop --show-labels]
[2] [oc] NAME                       READY   STATUS      RESTARTS   AGE   LABELS
[2] [oc] openjdk-app-app-1-deploy   0/1     Completed   0          80s   openshift.io/deployer-pod-for.name=openjdk-app-app-1
[2] [oc] openjdk-app-app-1-q6drm    1/1     Running     0          78s   deployment=openjdk-app-app-1,deploymentconfig=openjdk-app-app
[2] Running oc.exe with args [oc get dc openjdk-app-app --namespace hlongqcuop -o jsonpath='{range .spec.template.spec.containers[0].env[*]}{.name}:{.value}{"\n"}{end}']
[2] [oc] 'DEBUG_PORT:5858
[2] [oc] ODO_S2I_SCRIPTS_URL:/usr/local/s2i
[2] [oc] ODO_S2I_SCRIPTS_PROTOCOL:image://
[2] [oc] ODO_S2I_SRC_BIN_PATH:/tmp
[2] [oc] ODO_S2I_DEPLOYMENT_DIR:/deployments
[2] [oc] ODO_S2I_WORKING_DIR:/home/jboss
[2] [oc] ODO_S2I_BUILDER_IMG:redhat-[openjdk-18](https://issues.redhat.com/browse/openjdk-18)/openjdk18-openshift
[2] [oc] ODO_SRC_BACKUP_DIR:/opt/app-root/src-backup
[2] [oc] 'Running oc.exe with args [oc exec openjdk-app-app-1-q6drm --namespace hlongqcuop -- ls -lai \tmp\src]
[2] [oc] ls: cannot access \tmp\src: No such file or directory
[2] [oc] command terminated with exit code 2
[2] Deleting project: hlongqcuop
[2] Running odo.exe with args [odo project delete hlongqcuop -f]
[2] [odo] I1118 13:16:58.969738    9976 util.go:770] Cached response used.
[2] [odo]  V  Deleted project : hlongqcuop
[2] [odo]  !  Warning! Projects are deleted from the cluster asynchronously. Odo does its best to delete the project. Due to multi-tenant clusters, the project may still exist on a different 
node.
[2] [odo] I1118 13:16:58.997882    9976 odo.go:72] Could not get the latest release information in time. Never mind, exiting gracefully :)
[2] Setting current dir to: C:\Users\Admin\go\src\github.com\openshift\odo\tests\e2escenarios
[2] Deleting dir: C:\Users\Admin\AppData\Local\Temp\192507358```

### PR DESCRIPTION
**Pull request by anandrkskd**
Signed-off-by: anandrkskd <anandrkskd@gmail.com>

**What type of PR is this?**

/kind bug

**What does does this PR do / why we need it**:

This PR will Fix image source for E2E test image.

**Which issue(s) this PR fixes**:

Fixes #3389 
Fixes #4235 

**PR acceptance criteria**:

- [ ] Unit test 

- [ ] Integration test 

- [ ] Documentation 

- [x] I have read the [test guidelines](https://github.com/openshift/odo/blob/master/docs/dev/test-architecture.adoc)

**How to test changes / Special notes to the reviewer**:

### CODE REVIEW

File: tests/e2escenarios/e2e_images_test.go [hunk 1]
```diff
@@ -89,35 +89,35 @@ var _ = Describe("odo supported images e2e tests", func() {
 		})
 
 		It("Should be able to verify the nodejs-10 image", func() {
-			oc.ImportImageFromRegistry("registry.access.redhat.com", filepath.Join("rhoar-nodejs", "nodejs-10:latest"), "nodejs:latest", commonVar.Project)
-			verifySupportedImage(filepath.Join("rhoar-nodejs", "nodejs-10:latest"), "nodejs", "nodejs:latest", commonVar.Project, appName, commonVar.Context)
+			oc.ImportImageFromRegistry("registry.access.redhat.com", filepath.ToSlash(filepath.Join("rhoar-nodejs", "nodejs-10:latest")), "nodejs:latest", commonVar.Project)
```
Comment (kadel) [inline tests/e2escenarios/e2e_images_test.go:92]: You should not treat this as a file path. This is what got us into this situation in the first place.
Your solution would technically work, but it is still wrong and problematic. 
Just providing the whole image name as a string is completely ok here (`rhoar-nodejs/nodejs-10:latest`). The image name will always have `/` as a separator.

**Review by kadel — CHANGES_REQUESTED** [review]

File: tests/e2escenarios/e2e_images_test.go (hunk 1 above)
Comment (kadel) [reply tests/e2escenarios/e2e_images_test.go:92]: /hold 
please don't merge this before this gets properly fixed.

**Comment by kadel** [comment]
/hold
please don't merge this before this gets properly fixed. (see: https://github.com/openshift/odo/pull/4234#discussion_r525905756)

File: tests/e2escenarios/e2e_images_test.go (hunk 1 above)
Comment (anandrkskd) [reply tests/e2escenarios/e2e_images_test.go]: Understood

File: tests/e2escenarios/e2e_images_test.go [hunk 2]
```diff
@@ -89,35 +89,35 @@ var _ = Describe("odo supported images e2e tests", func() {
 		})
 
 		It("Should be able to verify the nodejs-10 image", func() {
-			oc.ImportImageFromRegistry("registry.access.redhat.com", filepath.Join("rhoar-nodejs", "nodejs-10:latest"), "nodejs:latest", commonVar.Project)
-			verifySupportedImage(filepath.Join("rhoar-nodejs", "nodejs-10:latest"), "nodejs", "nodejs:latest", commonVar.Project, appName, commonVar.Context)
+			oc.ImportImageFromRegistry("registry.access.redhat.com", "rhoar-nodejs"+"/"+"nodejs-10:latest", "nodejs:latest", commonVar.Project)
```
Comment (kadel) [inline tests/e2escenarios/e2e_images_test.go:92]: Why are you using `+` to concatenate two strings like this?

Comment (anandrkskd) [reply tests/e2escenarios/e2e_images_test.go:92]: I miss understood the above comment, Just providing the whole image name as a string is okay but in the helper function well will require to concatenate should I use string.join?

Comment (kadel) [reply tests/e2escenarios/e2e_images_test.go:92]: just do `"rhoar-nodejs/nodejs-10:latest"`

File: tests/helper/helper_oc.go [hunk 3]
```diff
@@ -312,8 +312,8 @@ func (oc OcRunner) checkForImageStream(name string, tag string) bool {
 
 // ImportImageFromRegistry import the required image of the respective component type from the specified registry
 func (oc OcRunner) ImportImageFromRegistry(registry, image, cmpType, project string) {
-	CmdShouldPass(oc.path, "--request-timeout", "5m", "import-image", cmpType, "--namespace="+project, "--from="+filepath.Join(registry, image), "--confirm")
-	CmdShouldPass(oc.path, "annotate", filepath.Join("istag", cmpType), "--namespace="+project, "tags=builder", "--overwrite")
+	CmdShouldPass(oc.path, "--request-timeout", "5m", "import-image", cmpType, "--namespace="+project, "--from="+registry+"/"+image, "--confirm")
+	CmdShouldPass(oc.path, "annotate", filepath.ToSlash(filepath.Join("istag", cmpType)), "--namespace="+project, "tags=builder", "--overwrite")
```
Comment (kadel) [inline tests/helper/helper_oc.go:316]: Don't use filepath if it is not file path. 

just do 
`fmt.Sprintf("istag/%s", cmpType)` or `"istag/"+ cmpType`

Comment (anandrkskd) [reply tests/helper/helper_oc.go:316]: Understood

**Pull request merged**